src/ZiraLink.Api.Application/Services/IdentityService.cs

```diff
@@ -0,0 +1,137 @@
+            {
+                Address = disco.TokenEndpoint,
+                ClientId = "back",
+                ClientSecret = "secret",
+                Scope = "ziralink IdentityServerApi"
+            }, cancellationToken);
+
+            if (tokenResponse.IsError)
+                throw new ApplicationException("Failed to get token from identity server");
+
+            httpClient.SetBearerToken(tokenResponse.AccessToken);
+
+            return httpClient;
+        }
+
+        public async Task<ApiResponse<string>> CreateUserAsync(string username, string password, string email, string name, string family, CancellationToken cancellationToken)
+        {
+            var httpClient = await InitializeHttpClientAsync(cancellationToken);
+
+             var jsonObject = new
+            {
+                Username = username,
+                Password = password,
+                Email = email,
+                Name = name,
+                Family = family
+            };
+
+            var content = new StringContent(JsonSerializer.Serialize(jsonObject), Encoding.UTF8, "application/json");
+            var baseUri = _idsUri;
+            var uri = new Uri(baseUri, "User");
+
+            HttpResponseMessage? response;
+
+            response = await httpClient.PostAsync(uri.ToString(), content);
+            ApiResponse<string> userCreationResult;
+            var responseString = await response.Content.ReadAsStringAsync(cancellationToken);
+
+            userCreationResult = JsonSerializer.Deserialize<ApiResponse<string>>(responseString, new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase }) ?? new ApiResponse<string>();
+
+            return userCreationResult;
+        }
+        
+        public async Task<ApiResponse<string>> ChangePasswordAsync(string userId, string currentPassword, string newPassword, CancellationToken cancellationToken)
+        {
+            var httpClient = await InitializeHttpClientAsync(cancellationToken);
+            
+             var jsonObject = new
+            {
+                UserId = userId,
+                CurrentPassword = currentPassword,
+                NewPassword = newPassword
+            };
+
+            var content = new StringContent(JsonSerializer.Serialize(jsonObject), Encoding.UTF8, "application/json");
+            var baseUri = _idsUri;
+            var uri = new Uri(baseUri, "User/ChangePassword");
+
+            HttpResponseMessage? response;
+
+            response = await httpClient.PatchAsync(uri.ToString(), content);
+            ApiResponse<string> userCreationResult;
+            var responseString = await response.Content.ReadAsStringAsync(cancellationToken);
+
+            userCreationResult = JsonSerializer.Deserialize<ApiResponse<string>>(responseString, new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase }) ?? new ApiResponse<string>();
```

Comment:
userCreationResult should be renamed to changePasswordResult

Reply:
Done.